rust/plugin-lib/src/view.rs

```diff
@@ -131,7 +131,10 @@ impl<C: Cache> View<C> {
         self.peer.send_rpc_notification("add_scopes", &params);
     }
 
-    pub fn edit(&self, edit: PluginEdit) {
+    pub fn edit(&self, delta: RopeDelta, priority: u64, after_cursor: bool,
+                undo_group: Option<usize>, author: String) {
```

Comment:
This is just about right, but one little change: instead of `undo_group: Option<usize>` I'd have an argument `new_undo_group: bool`; then we just keep track of the undo group internally, like we do with `rev`, and send it to core whenever `new_undo_group` is `false`.